src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskEntityToCamundaReconfigureInputDefMapperTest.java

```diff
@@ -0,0 +1,126 @@
+import uk.gov.hmcts.reform.wataskmanagementapi.entity.TaskRoleResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.entity.WorkTypeResource;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.singleton;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static uk.gov.hmcts.reform.wataskmanagementapi.cft.enums.CFTTaskState.COMPLETED;
+
+class TaskEntityToCamundaReconfigureInputDefMapperTest {
+    private final TaskEntityToCamundaReconfigureInputDefMapper mapper =
+        Mappers.getMapper(TaskEntityToCamundaReconfigureInputDefMapper.class);
+    private TaskResource taskResource;
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    private CamundaReconfigureInputVariableDefinition camundaReconfigureInputVarDef;
+
+    public static final Map<String, String> EXPECTED_ADDITIONAL_PROPERTIES = Map.of(
+        "name1", "value1",
+        "name2", "value2"
+    );
+
+    @BeforeEach
+    void setup() {
+        this.objectMapper.registerModule(new JavaTimeModule());
+        TaskRoleResource roleResource = new TaskRoleResource(
+            "tribunal-caseworker",
+            true,
+            true,
+            true,
+            true,
+            true,
+            true,
+            new String[]{"SPECIFIC", "STANDARD"},
+            0,
+            false,
+            "JUDICIAL",
+            "taskId",
+            OffsetDateTime.parse("2021-05-09T20:15:45.345875+01:00")
+        );
+        taskResource = new TaskResource(
+            "taskId",
+            "aTaskName",
+            "startAppeal",
+            OffsetDateTime.parse("2022-05-09T20:15:45.345875+01:00"),
+            COMPLETED,
+            TaskSystem.SELF,
+            SecurityClassification.PUBLIC,
+            "title",
+            "a description",
+            null,
+            0,
+            0,
+            "someAssignee",
+            false,
+            new ExecutionTypeResource(ExecutionType.MANUAL, "Manual", "Manual Description"),
+            new WorkTypeResource("routine_work", "Routine work"),
+            "JUDICIAL",
+            false,
+            OffsetDateTime.parse("2022-05-09T20:15:45.345875+01:00"),
+            "1623278362430412",
+            "Asylum",
+            "TestCase",
+            "IA",
+            "1",
+            "TestRegion",
+            "765324",
+            "Taylor House",
+            BusinessContext.CFT_TASK,
+            null,
+            OffsetDateTime.parse("2021-05-09T20:15:45.345875+01:00"),
+            singleton(roleResource),
+            "caseCategory",
+            EXPECTED_ADDITIONAL_PROPERTIES,
+            "nextHearingId",
+            OffsetDateTime.parse("2021-05-09T20:15:45.345875+01:00"),
+            OffsetDateTime.parse("2021-05-09T20:15:45.345875+01:00")
+        );
```

Comment:
Probably outside the scope of this but we should have a better way of creating test objects so it is re-usable and easier to read.  Leave this for now but something we ought to refactor in future.